src/main/kotlin/com/kotlindiscord/site/routes/api/Infractions.kt

```diff
@@ -0,0 +1,109 @@
+package com.kotlindiscord.site.routes.api
+
+import com.kotlindiscord.api.client.models.InfractionFilterModel
+import com.kotlindiscord.api.client.models.InfractionModel
+import com.kotlindiscord.database.*
+import com.kotlindiscord.site.components.apiRoute
+import com.kotlindiscord.site.models.fromDB
+import io.ktor.application.call
+import io.ktor.http.HttpStatusCode
+import io.ktor.request.receive
+import io.ktor.response.respond
+import org.jetbrains.exposed.sql.and
+import org.jetbrains.exposed.sql.andWhere
+import org.jetbrains.exposed.sql.or
+import org.jetbrains.exposed.sql.selectAll
+import org.jetbrains.exposed.sql.transactions.experimental.newSuspendedTransaction
+import java.time.Instant
+import java.time.LocalDateTime
+
+val apiInfractionsGet = apiRoute {
+    val model = call.receive<InfractionFilterModel>()
+
+    if (model.createdAfter != null && model.createdBefore != null) {
+        call.respond(
+            HttpStatusCode.NotAcceptable,
+            mapOf("error" to "Specify one of createdAfter and createdBefore, not both.")
+        )
+
+        return@apiRoute null
+    }
```

Comment:
Is there any reason why both can't be provided? It seems like a pretty reasonable query to want to make - e.g all infractions created during Pride Month 2020.

Reply:
Oh, good point actually